service/awsconfig/framework.go

```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
 		}
 	}
 
-	// We provide a map of resource lists keyed by the version bundle version
-	// to the resource router.
-	versionedResources := map[string][]framework.Resource{
-		// Clusters without a version use the legacy resource.
-		"":      legacyResources,
-		"0.1.0": legacyResources,
-		"0.2.0": resourcesV2,
-		"1.0.0": legacyResources,
-		"2.0.0": resourcesV2,
-		"2.0.1": resourcesV2_0_1,
-		// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
-		"2.0.2": resourcesV2_0_1,
+	handlesFuncFunc := func(handledVersionBundles []string) func(obj interface{}) bool {
+		return func(obj interface{}) bool {
+			kvmConfig, err := key.ToCustomObject(obj)
+			if err != nil {
+				return false
+			}
+			versionBundleVersion := key.VersionBundleVersion(kvmConfig)
+
+			for _, v := range handledVersionBundles {
+				if versionBundleVersion == v {
+					return true
+				}
+			}
+
+			return false
+		}
+	}
+
+	var resourceSetV1 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"",
+				"0.1.0",
+				"1.0.0",
+			}),
+			Logger:    config.Logger,
+			Resources: legacyResources,
+		}
+
+		resourceSetV1, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceSetV2 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"0.2.0",
+				"2.0.0",
+			}),
+			Logger:    config.Logger,
+			Resources: resourcesV2,
+		}
+
+		resourceSetV2, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceSetV4 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"2.0.1",
+				// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
+				"2.0.2",
+			}),
+			Logger:    config.Logger,
+			Resources: resourcesV2_0_1,
+		}
+
+		resourceSetV4, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceRouter *framework.ResourceRouter
+	{
+		c := framework.ResourceRouterConfig{}
+
+		c.ResourceSets = []*framework.ResourceSet{
+			resourceSetV1,
+			resourceSetV2,
+			resourceSetV4,
```

Comment:
As you can see there is no v3, I'll remove the directory later after copying stuff to v4.

Reply:
Why is this though? This seems to be super wrong.

Reply:
I think we got out of sync due to the CRD  / CF migration. e.g we had 

- cloudconfigv1 - TPR
- cloudconfigv2 - CRD
- cloudconfigv3 - CRD + k8scloudconfig v_3_0_0 for CloudFormation

Could we have resourceSetV3 supporting bundles `2.0.1` and `2.0.2`? That makes more sense to me.

Reply:
I think naming tired to reflect version bundle versions and that's why we have situation like that. But we don't have to worry, as the new structure is there to improve.

Reply:
I would flatten v3 and v4. WDYT?

Reply:
I think flattening is a good idea. The problem will go away as we delete old resource sets but in the short term it's confusing.

Reply:
Renamed to resourceSetV3
I'll flatten v4 to v3 in follow up PRs.